sound/soc/sof/ipc4-pcm.c

```diff
@@ -504,7 +504,7 @@ static int sof_ipc4_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 		snd_mask_set_format(fmt, SNDRV_PCM_FORMAT_S32_LE);
 	}
 
-	rate->min = ipc4_copier->available_fmt.input_audio_fmts->sampling_frequency;
+	rate->min = ipc4_copier->available_fmt.input_audio_fmts->audio_fmt.sampling_frequency;
```

Comment:
I am not able to follow the original code or the suggested change. In both cases, input_audio_fmts can be an array of "@audio_fmt_num: Number of available audio formats" elements, so why take the first element only?

in other places you use &available_fmt->input_audio_fmts[i].audio_fmt, so not sure why the index 0 is used?

Reply:
this was added by @RanderWang in bed4045c81cd70. Can you comment Rander?

Reply:
Erm, what @RanderWang? @kv2019i is precisely trying to add more rates to the BT pipelines, so I am not sure how your assumption that dai copiers have a single sampling rate could hold

Reply:
yes, my assumption is stale.